user-interface/src/main/java/life/qbic/datamanager/views/projects/project/ontology/OntologyLookupComponent.java

```diff
@@ -0,0 +1,235 @@
+    ontologyGrid.addClassName("ontology-grid");
+    ontologyGrid.addThemeVariants(GridVariant.LUMO_NO_BORDER, GridVariant.LUMO_NO_ROW_BORDERS);
+    setLazyDataProviderForOntologyGrid(ontologyGrid);
+    ontologyGridSection.add(foundResults, ontologyGrid);
+    foundResults.addClassName("secondary");
+    ontologyGridSection.addClassName("ontology-grid-section");
+    add(ontologyGridSection);
+  }
+
+  private void updateShownResults(OntologySearchEvent event) {
+    searchTerm = event.searchValue();
+    ontologyGridLazyDataView.refreshAll();
+    foundResults.setText("%s results found".formatted(ontologyGridLazyDataView.getItems().count()));
+    if (!ontologyGridSection.isVisible()) {
+      ontologyGridSection.setVisible(true);
+    }
+  }
+
+  /**
+   * Resets the value within the searchfield, which in turn resets the grid via the fired
+   * {@link OntologySearchEvent}. Additionally, hides the entire section so the result span is only
+   * shown when the user is actively searching for an ontology
+   */
+  public void resetSearch() {
+    searchField.setValue("");
+    ontologyGridSection.setVisible(false);
+  }
+
+  /**
+   * Register an {@link ComponentEventListener} that will get informed with a
+   * {@link OntologySearchEvent}, as soon as a user searches for an ontology of interest.
+   *
+   * @param ontologySearchListener a listener on the search event trigger
+   */
+  public void addSearchEventListener(
+      ComponentEventListener<OntologySearchEvent> ontologySearchListener) {
+    addListener(OntologySearchEvent.class, ontologySearchListener);
+  }
+
+  private void initSearchField() {
+    searchField.setClassName("search-field");
+    searchField.setPlaceholder("Search for ontology term e.g. Whole blood");
+    searchField.setHelperText("Please provide at least %s letters to search for entries".formatted(
+        ONTOLOGY_SEARCH_LOWER_LIMIT));
+    searchField.setPrefixComponent(VaadinIcon.SEARCH.create());
+    searchField.setValueChangeMode(ValueChangeMode.LAZY);
+    searchField.addValueChangeListener(
+        event -> {
+          if (event.getValue().length() >= ONTOLOGY_SEARCH_LOWER_LIMIT) {
+            fireEvent(new OntologySearchEvent(this, event.isFromClient(), event.getValue()));
+          }
+        });
+    add(searchField);
+  }
+
+
+  /**
+   * Ontology Item
+   * <p>
+   * The ontology Item is a Div container styled similiar to the {@link Card} component, hosting the
+   * ontology information of interest provided by a {@link OntologyClassDTO}
+   */
+  private static class OntologyItem extends Div {
+
+    public OntologyItem(String label, String name, String classIri, String description,
+        String ontologyAbbreviation) {
+      addHeader(label, name);
+      addUrl(classIri);
+      addDescription(description);
+      addOrigin(ontologyAbbreviation);
+      addClassName("ontology-item");
+    }
+
+    private void addHeader(String label, String nameText) {
```

Comment:
rather than adding in different methods to this object, please return the objects to be added and compose the component in its constructor. It can be hard to find modifications of the tree otherwise.